job-server/src/spark.jobserver/LocalContextSupervisorActor.scala

```diff
@@ -126,7 +128,16 @@ class LocalContextSupervisorActor(dao: JobDAO) extends InstrumentedActor {
 
     case GetContext(name) =>
       if (contexts contains name) {
-        sender ! (contexts(name), resultActors(name))
+        val future = (contexts(name) ? SparkContextStatus) (contextTimeout.seconds)
+        Await.result(future, contextTimeout.seconds) match {
```

Comment:
Instead of using `Await.result`, which blocks the thread, it's better to map the future:
```
  future.collect {
    case SparkContextAlive =>   sender ! (contexts(name), resultActors(name))
    case SparkContextDead =>
      self ! StopContext(name)
      sender ! NoSuchContext
  }
```

Also:  note that  { } are not needed after a case, and it would be really good to log when the context was found to be dead.